compiler/hash-reporting/src/report.rs

```diff
@@ -1,28 +1,34 @@
-//! Hash Compiler error and warning reporting module.
-use crate::highlight::{highlight, Colour, Modifier};
-use core::fmt;
-use hash_error_codes::error_codes::HashErrorCode;
-use hash_source::{location::SourceLocation, SourceMap};
-use hash_utils::path::adjust_canonicalization;
 use std::{
     cell::Cell,
+    fmt,
     iter::{once, repeat},
 };
 
-const ERROR_MARKING_CHAR: &str = "^";
+use hash_error_codes::error_codes::HashErrorCode;
+use hash_source::{location::SourceLocation, SourceMap};
+use hash_utils::path::adjust_canonicalization;
+
+use crate::{
+    highlight::{highlight, Colour, Modifier},
+    writer::{offset_col_row, ReportCodeBlockInfo, ERROR_MARKING_CHAR},
+};
 
-/// Enumeration describing the type of report; either being a warning, info or a
+/// Enumeration describing the kind of [Report]; either being a warning, info or a
 /// error.
 #[derive(Debug, Copy, Clone, Hash, Eq, PartialEq)]
 pub enum ReportKind {
+    /// The report is an error.
     Error,
+    /// The report is an informational diagnostic(likely for internal purposes).
     Info,
+    /// The report is a warning.
     Warning,
+    // @@TODO: Add `Internal` variant: This is an internal compiler error.
```

Comment:
Does this mean that panics essentially create Internal reports and print them when they happen?

Reply:
But is the reporter responsible for the abort? Surely it would make more sense if as far as the reporter is concerned Internal  is just another "color mode" (anyway, a cosmetic setting); and the actual abort logic is basically just the panic handler creating an Internal report, printing it, and then aborting right there and then (so maybe the panic handler should add the stack trace as part of the report).

Reply:
Yes that is what I was intending, it's purely cosmetic and the panic should occur in the call site.